HipercowApiUnitTests/Controllers/ClusterLoadControllerUnitTest.cs

```diff
@@ -0,0 +1,97 @@
+// Copyright (c) Imperial College London. All rights reserved.
+
+namespace HipercowApiUnitTests.Controllers
+{
+    using HipercowApi.Controllers;
+    using HipercowApi.Models;
+    using HipercowApi.Tools;
+    using Microsoft.Hpc.Scheduler;
+    using Microsoft.Hpc.Scheduler.Properties;
+    using Moq;
+
+    /// <summary>
+    /// Test the /clusterload endpoint.
+    /// </summary>
+    public class ClusterLoadControllerUnitTest
+    {
+        /// <summary>
+        /// Test the /clusterload/cluster
+        /// by providing fake data.
+        /// </summary>
+        [Fact]
+        public void GetClusterLoad_Works()
+        {
+            SchedulerCoreState busy = SchedulerCoreState.Busy;
+            SchedulerCoreState idle = SchedulerCoreState.Idle;
+            SchedulerCoreState offline = SchedulerCoreState.Offline;
+            NodeState n_online = NodeState.Online;
+            NodeState n_offline = NodeState.Offline;
+
+            SchedulerCollection<ISchedulerNode> mockNodeList =
+            [
+                FakeNode("node-1", FakeCores([busy, idle, offline]), n_online),
+                FakeNode("node-2", FakeCores([busy, idle]), n_offline),
+            ];
+
+            var mockScheduler = new Mock<IScheduler>();
+            mockScheduler.Setup(x => x.Connect("potato")).Verifiable();
+            mockScheduler.Setup(x => x.GetNodeList(
+                It.IsAny<IFilterCollection>(),
+                It.IsAny<ISortCollection>())).Returns(mockNodeList);
+
+            var mockHandleCache = new Mock<IClusterHandleCache>();
+            mockHandleCache.Setup(x => x.GetClusterHandle("potato")).
+                                         Returns(mockScheduler.Object);
+
+            var cc = new ClusterLoadController(
+                new ClusterLoadQuery(),
```

Comment:
OK, so you're unit testing `ClusterLoadController` and `ClusterLoadQuery` together here. Technically these ought to be unit tested separately, which would make it easier to test fiddly things like does the cluster load query exclude the head node (which I don't think you're doing here...?) But since this branch has already been round the houses a bit, you might not want to do that right away, but I think it would be good to at least put a comment here noting the fact that there aren't any dedicated ClusterNodeQuery unit tests yet.

Reply:
Yeah - I guess I've bundled them together at the moment - imilar to `ClusterInfoQuery` which isn't tested on its own, but only via the call to it from ClustersController. The controllers are pretty minimal functino wrappers for both...

I'll save that for the next PR!